src/main/java/randoop/generation/constanttfidf/TfIdfSelector.java

```diff
@@ -0,0 +1,140 @@
+package randoop.generation.constanttfidf;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import randoop.main.RandoopBug;
+import randoop.sequence.Sequence;
+import randoop.util.Log;
+import randoop.util.Randomness;
+import randoop.util.list.SimpleList;
+
+/**
+ * This class selects a sequence based on TF-IDF. TfIdfSelector is only used when {@code
+ * --constant-tfidf} is enabled.
+ *
+ * <p>There is one TfIdfSelector per scope. When the literal level is ClassOrInterfaceType or
+ * Package, a scope is a class or a package, respectively. When the literal level is ALL, there is
+ * one global TfIdfSelector.
+ *
+ * <p>By information, it means sequence numUses and number of occurrence.
+ */
+public class TfIdfSelector {
+
+  /** If true, output debugging information. */
+  private static final boolean DEBUG = false;
+
+  /** Map from a sequence to its TF-IDF weight. */
+  Map<Sequence, Double> constantWeight = new HashMap<>();
+
+  /**
+   * Create a TfIdfSelector.
+   *
+   * <p>The two maps that are passed in have the same keyset.
+   *
+   * @param numUses map from sequence to its number of uses (in the represented scope)
+   * @param classesWithConstant map from sequence to the number of classes (in the represented
+   *     scope) that contain the sequence; null if the literal level is CLASS
+   * @param classCount the total number of classes (in the represented scope)
+   */
+  public TfIdfSelector(
+      Map<Sequence, Integer> numUses,
+      @Nullable Map<Sequence, Integer> classesWithConstant,
+      int classCount) {
+    if (DEBUG) {
+      Log.logPrintf(
+          "Initializing TF-IDF Selector: %n"
+              + "Sequence numUses: "
+              + numUses
+              + "%n"
+              + "Sequence occurrence: "
+              + classesWithConstant
+              + "%n"
+              + "Class count: "
+              + classCount
+              + "%n");
+    }
+    if (numUses.isEmpty()) {
+      Log.logPrintf("TF-IDF Selector: Sequence numUses is empty");
+      return;
+    }
+
+    if (!numUses.keySet().equals(classesWithConstant.keySet())) {
```

Comment:
`classesWithConstant` has a `@Nullable` type qualifier, but here it is dereferenced unconditionally.  Is the specification wrong, or is the code wrong?